src/types/index.ts

```diff
@@ -345,18 +358,22 @@ export class StarknetContractFactory {
     public abiPath: string;
     private constructorAbi: starknet.CairoFunction;
     public metadataPath: string;
+    public casmPath: string;
     private classHash: string;
 
     constructor(config: StarknetContractFactoryConfig) {
         this.hre = config.hre;
         this.abiPath = config.abiPath;
         this.abi = readAbi(this.abiPath);
         this.metadataPath = config.metadataPath;
+        if (config.casmPath) {
```

Comment:
Why is the `if` necessary? Can't we just do `this.casmPath = config.casmPath`?

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/328/commits/4e34b4ac19a48f2c9c22f8967a5f7c80abb3241c